WIP Issue#127 ConfigureQuartz. Added a basic working configuration. Added…

Dandria1802 opened the pull request.

Dandria1802 committed:
Issue#127 ConfigureQuartz. Added a basic working configuration. Added tables for store QuartzData.

Dandria1802 committed:
Update application.properties

Dandria1802 committed:
Update QuartzConfig.java

Dandria1802 committed:
Update UserRepository.java

kh0ma commented on home-application/src/main/resources/config/tables_postgres.sql:
```diff
@@ -0,0 +1,187 @@
+-- Thanks to Patrick Lightbody for submitting this...
```
Can we get rid of this sql file?

Dandria1802 replied:
Yes.

kh0ma commented on home-application/src/main/resources/config/application.properties:
```diff
@@ -1,3 +1,8 @@
+spring.quartz.job-store-type = jdbc
+spring.quartz.properties.org.quartz.jobStore.class=org.quartz.impl.jdbcjobstore.JobStoreCMT
+spring.quartz.properties.org.quartz.jobStore.driverDelegateClass=org.quartz.impl.jdbcjobstore.PostgreSQLDelegate
+spring.quartz.jdbc.initialize-schema=always
```
What other options do we have?

kh0ma commented on home-application/src/main/java/com/softserveinc/ita/homeproject/application/quartz/triggers/DataRetrievalTrigger.java:
```diff
@@ -0,0 +1,16 @@
+public class DataRetrievalTrigger {
+    @Bean(name = "dataRetrievalTriggers")
+    public SimpleTriggerFactoryBean triggerMemberStats(@Qualifier("dataRetrieval") JobDetail jobDetail) {
+        return QuartzConfig.createTrigger(jobDetail, 60000, "Data retrieval job trigger");
```
Is there any example of a particular job? It isn't clear from naming. I would like to see how for instance notifications job will be set.

Dandria1802 replied:
I will change the class names. These are really incomprehensible names.

Dandria1802 replied:
Yes, I'll make a working example.

Dandria1802 committed:
Delete tables_postgres.sql

An unnecessary file has been deleted.

Dandria1802 committed:
Issue#127 ConfigureQuartz
The structure of the classes has been changed. Added liquibase script for Quartz.

Dandria1802 committed:
Update Issue127-QuartzConfig.xml

Added new line

PoLytvy committed:
Merge branch 'Issue#127-Implementation-of-notification' into Issue#127-Quartz-Configuration

# Conflicts:
#	home-application/src/main/resources/config/application.properties

PoLytvy committed:
Issue#127: Merge conflicts
